packages/drivers/odsp-driver/src/odspDeltaStorageService.ts

```diff
@@ -162,6 +164,8 @@ export class OdspDeltaStorageWithCache implements IDocumentDeltaStorageService {
 		// Better implementation would be to return only what we have in cache, but that also breaks API
 		assert(!cachedOnly || toTotal === undefined, 0x1e3);
 
+		// Don't use cache for ops is snapshot is fetched from network or if it was not fetched at all.
+		this.useCacheForOps = this.storageManagerGetter()?.isFirstSnapshotFromNetwork === false;
```

Comment:
This will reset this.useCacheForOps every time we call fetchMessages() API.
I think we want 

this.useCacheForOps = this.useCacheForOps && (this.storageManagerGetter()?.isFirstSnapshotFromNetwork === false)